src/swarm/neo/client/mixins/DeserializeMethod.d

```diff
@@ -0,0 +1,73 @@
+/*******************************************************************************
+
+    Mixin for method to deserialize a received record.
+
+    Copyright:
+        Copyright (c) 2017 sociomantic labs GmbH. All rights reserved
+
+    License:
+        Boost Software License Version 1.0. See LICENSE.txt for details.
+
+*******************************************************************************/
+
+module swarm.neo.client.mixins.DeserializeMethod;
+
+import ocean.util.serialize.contiguous.MultiVersionDecorator;
+
+/*******************************************************************************
+
+    Mixin for method to deserialize a received record.
+
+    Params:
+        src = the buffer to be deserialized
+
+*******************************************************************************/
+
+template DeserializeMethod ( alias src )
+{
+    import ocean.util.serialize.contiguous.Contiguous;
+    import ocean.util.serialize.Version;
+    import ocean.util.serialize.contiguous.Deserializer;
+    import ocean.util.serialize.contiguous.MultiVersionDecorator;
+
+    /***************************************************************************
+
+        Deserializes `src` into `dst`, using the provided version decorater, if
+        required.
+
+        Params:
+            T = type of struct to deserialize
+            dst = deserialization destination
+
+        Returns:
+            deserialized struct
+
+    ***************************************************************************/
+
+    public Contiguous!(T) deserialize ( T ) ( ref Contiguous!(T) dst )
+    {
+        static if ( Version.Info!(T).exists )
+        {
+            if ( version_decorator is null )
+                version_decorator = new VersionDecorator;
+
+            return version_decorator.loadCopy!(T)(src, dst);
+        }
+        else
+        {
+            return Deserializer.deserialize(src, dst);
+        }
+    }
+}
+
+/*******************************************************************************
+
+    VersionDecorator singleton used by all mixed-in deserialize() methods when
+    handling versioned structs.
+
+    Must be public as it needs to be accessible from the point where the
+    deserialize method is mixed in. Not intended to be used externally by users.
+
+*******************************************************************************/
+
+public VersionDecorator version_decorator;
```

Comment:
As it is public, I'd recommend to use more obfuscated long name, like `shared_client_version_decorator`.